File: bioedge-backend/src/utils/email_service.py
import datetime
import sys

# Ensure UTF-8 output on Windows consoles
if sys.stdout and hasattr(sys.stdout, "reconfigure"):
    try:
        sys.stdout.reconfigure(encoding="utf-8")
    except Exception:
        pass
if sys.stderr and hasattr(sys.stderr, "reconfigure"):
    try:
        sys.stderr.reconfigure(encoding="utf-8")
    except Exception:
        pass

class EmailService:
    @staticmethod
    def send_verification_code(to_email: str, code: str, recipient_name: str = "Student") -> bool:
        """
        Deliver 6-digit verification code.
        Logs a rich visual email receipt to the server terminal.
        """
        now_str = datetime.datetime.now().strftime("%I:%M %p")
        
        email_template = f"""
======================================================
📧 [BIO EDGE EMAIL SERVICE] Verification Code Delivery
To: {recipient_name} <{to_email}>
Subject: Verify Your Email - Bio Edge by Afroza Tahmina
Time: {now_str}
------------------------------------------------------
Dear {recipient_name},
Your 6-digit Bio Edge verification code is:

       >>>  [ {code} ]  <<<

This code will expire in 10 minutes.
If you did not request this, please disregard this email.
======================================================
"""
        print(email_template)
        return True

File: bioedge-backend/src/utils/emailService.js
/**
 * Email Service for Bio Edge Platform
 * Handles 6-digit OTP verification code delivery and notifications.
 */

export const EmailService = {
  /**
   * Send 6-digit verification code to the student's email
   * @param {string} email 
   * @param {string} code 
   * @param {string} name 
   */
  async sendVerificationCode(email, code, name = 'Student') {
    const formattedDate = new Date().toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit' });

    console.log('\n======================================================');
    console.log(`📧 [BIO EDGE EMAIL SERVICE] Verification Code Delivery`);
    console.log(`To: ${name} <${email}>`);
    console.log(`Subject: Verify Your Email - Bio Edge by Afroza Tahmina`);
    console.log(`Time: ${formattedDate}`);
    console.log(`------------------------------------------------------`);
    console.log(`Dear ${name},`);
    console.log(`Your 6-digit Bio Edge verification code is:`);
    console.log(`\n       >>>  [ ${code} ]  <<<\n`);
    console.log(`This code will expire in 10 minutes.`);
    console.log(`If you did not request this, please disregard this email.`);
    console.log('======================================================\n');

    return {
      success: true,
      deliveredTo: email,
      code
    };
  }
};
